test/unit/master/test_build.py

```diff
@@ -563,13 +582,16 @@ def _create_mock_slave(self, num_executors=5):
 
         return mock_slave
 
-    def _create_mock_subjob_calc(self, subjobs):
+    def _create_mock_subjob_calc(self, subjobs, compute_subjob_raises_exception=False):
         """
         :type subjobs: list[Subjob]
         :rtype: SubjobCalculator
         """
         mock_subjob_calculator = MagicMock(spec_set=SubjobCalculator)
-        mock_subjob_calculator.compute_subjobs_for_build.return_value = subjobs
+        if compute_subjob_raises_exception:
+            mock_subjob_calculator.compute_subjobs_for_build.side_effect = [AtomizerError('Atomizer command failed!')]
```

Comment:
the rhs doesn't need to be an array. You can just do 
`= AtomizerError('Atomizer command failed!')`
or
`= AtomizerError`